android/build.gradle

```diff
@@ -192,3 +192,79 @@ dependencies {
         }
     }
 }
+
+def isScreensExampleApp() {
+    return project.hasProperty('isScreensExampleApp') && project.property('isScreensExampleApp')
+}
+
+def getAbsoluteCodegenArtifactsPaperDestination() {
+    if (!project.hasProperty('codegenArtifactsPaperDestination')) {
+        throw new Exception('[RNScreens] Please fill codegenArtifactsPaperDestination variable in android/gradle.properties correct path to paper paper destination')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsPaperDestination')}"
+}
+
+def getAbsoluteCodegenArtifactsSource() {
+    if (!project.hasProperty('codegenArtifactsSource')) {
+        throw new Exception('[RNScreens] Please fill codegenArtifactsSource variable in android/gradle.properties correct path to codegenerated artifacts')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsSource')}"
+}
+
+
+tasks.register("copyCodegenArtifacts") {
+    group 'After build tasks'
+    description 'Tasks which copy codegen artifacts to paper architecture'
+    
+    if (!isScreensExampleApp() || !isNewArchitectureEnabled()) {
+        return
+    }
+
+    dependsOn tasks.generateCodegenArtifactsFromSchema
+
+    doLast {
+
+        def absoluteCodegenArtifactsPaperDestination = getAbsoluteCodegenArtifactsPaperDestination()
+        def absoluteCodegenArtifactsSource = getAbsoluteCodegenArtifactsSource()
+
+        def existingFiles = fileTree(absoluteCodegenArtifactsPaperDestination).matching {
+            include '**/*.java'
+        }
+
+        def generatedFiles = fileTree(absoluteCodegenArtifactsSource).matching {
+            include '**/*.java'
+        }
+
+        def existingFilesMap = [:]
+
+        existingFiles.forEach { existingFile ->
+            existingFilesMap[existingFile.name] = 1
+        }
+
+        generatedFiles.forEach { generatedFile -> 
+            if (!existingFilesMap.containsKey(generatedFile.name)) {
+                println("[RNScreens] ${generatedFile.name} not found in paper dir, if it's used in Android you need to copy it manually and implement yourself before using auto-copy feature")
+            }
+        }
+
+        if (existingFiles.size() == 0) {
+            throw new Exception("[RNScreens] Paper destination with codegen interfaces is empty. This might be okay if you don't have any interfaces/delegates used in Android, if that's not the case please check if codegenArtifactsPaperDestination in android/gradle.properties is correct")
```

Comment:
If there are cases when this is ok, we should print a warning instead of throwing an exception.

Reply:
fixed in https://github.com/software-mansion/react-native-screens/pull/2168/commits/9a2a66e36f40345e800cb005bd69ce1827054051